ray_provider/triggers.py

```diff
@@ -111,46 +114,42 @@ async def _stream_logs(self) -> None:
 
     async def run(self) -> AsyncIterator[TriggerEvent]:
         """
-        Asynchronously polls the job status and yields events based on the job's state.
+        Asynchronously polls the Ray job status and yields events based on the job's state.
 
         This method gets job status at each poll interval and streams logs if available.
         It yields a TriggerEvent upon job completion, cancellation, or failure.
 
         :yield: TriggerEvent containing the status, message, and job ID related to the job.
         """
-        try:
-            self.log.info(f"Polling for job {self.job_id} every {self.poll_interval} seconds...")
+        self.log.info(f"::group:: Trigger 1/2: Checking the job status")
+        self.log.info(f"Polling for job {self.job_id} every {self.poll_interval} seconds...")
 
+        try:
             tasks = [self._poll_status()]
             if self.fetch_logs:
                 tasks.append(self._stream_logs())
-
             await asyncio.gather(*tasks)
+        except ApiException as e:
+            error_msg = str(e)
+            self.log.info(f"::endgroup::")
+            self.log.error("::group:: Trigger unable to poll job status")
+            self.log.error("Exception details:", exc_info=True)
+            self.log.info("Attempting to clean up...")
+            await self.cleanup()
```

Comment:
wondering what happens when someone marks an Airflow task as failed from the Airflow UI when it is in the deferred state does this self.cleanup() get executed?

Reply:
That's a good question, @pankajastro , it will not. Do you know what exception is raised by Airflow in those cases? Do you think the triggerer the right place to handle this?

Reply:
If we can do it from operator `execute_complete` I think that will be better, wdyt?

Reply:
We can also capture this one, I'll add it!

Reply:
@pankajastro I marked the task in the UI as failed, while it was in the `(SubmitJob 4/5) Wait for completion` state, and the RayCluster was deleted.

I believe this is being handled from within the operator or hook code. I will have more follow up PRs removing these exceptions - would it be okay if we confirm that after all these PRs the behaviour continue as expected - before we release 0.3.0, as part of #77?

Reply:
sure